cpp/CMakeLists.txt

```diff
@@ -487,11 +499,16 @@ if(NOT BUILD_CPU_ONLY)
                                  ${CUVS_CUSPARSE_DEPENDENCY} ${CUVS_CURAND_DEPENDENCY}
   )
 
+  if(BUILD_MG_ALGOS)
+    set(CUVS_COMMS_DEPENDENCY ucp ucs ucxx nccl)
+  endif()
+
   # Keep cuVS as lightweight as possible. Only CUDA libs and rmm should be used in global target.
   target_link_libraries(
     cuvs
     PUBLIC rmm::rmm raft::raft ${CUVS_CTK_MATH_DEPENDENCIES}
     PRIVATE nvidia::cutlass::cutlass $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
+            ${CUVS_COMMS_DEPENDENCY}
```

Comment:
Have you tried building with `BUILD_MG_ALGOS=OFF`? I wonder if CMake will throw an error saying that this variable is undefined.

Also, we want to add `if (BUILD_MG_ALGOS) target_compile_options(cuvs CUVS_BUILD_MG_ALGOS) endif()` so that we can optionally compile the header and throw an error if a user tries to include `ann_mg.hpp` header with `BUILD_MG_ALGOS=OFF`

Reply:
Setting `BUILD_MG_ALGOS=OFF` does not seem to throw an error. Yes, just added a compile definition and exception check.